### ISSUE TITLE
reloading on worker processes

### ISSUE THREAD

**Issue by shashi** [timholy/Revise.jl#147]
It would be AWESOME to get Revise to work with packages that are intended to work with multiple worker processes. Currently, it seems to have problems referencing anonymous functions across workers.

Steps to reproduce:

1. Create a package called `ParReviseExample`

```julia
module ParReviseExample
using Distributed

greet(x) = println("Hello, ", x)

foo() = for p in workers()
    remotecall_fetch(() -> greet("Bar"), p)
end

end # module
```

2. Load a couple of workers, and run `ParReviseExample.foo()`
```julia
julia> using Distributed

julia> addprocs(2)
2-element Array{Int64,1}:
u 2
 3

julia> using ParReviseExample
## ignore warnings ... :-p

julia> ParReviseExample.foo()
      From worker 2:    Hello, Bar
      From worker 3:    Hello, Bar
```
3. Apply this patch:
```julia
index 02f3e08..b50eb7f 100644
--- a/src/ParReviseExample.jl
+++ b/src/ParReviseExample.jl
@@ -4,7 +4,7 @@ using Distributed
 greet(x) = println("Hello, ", x)
 
 foo() = for p in workers()
-    remotecall_fetch(() -> greet("Bar"), p)
+    remotecall_fetch((x) -> greet("Bar"), p, 1)
 end
 
 end # module
```
4. try again

```julia
julia> ParReviseExample.foo()
ERROR: On worker 2:
UndefVarError: ##3#4 not defined
deserialize_datatype at /home/shashi/code/julia7/usr/share/julia/stdlib/v0.7/Serialization/src/Serialization.jl:1053
handle_deserialize at /home/shashi/code/julia7/usr/share/julia/stdlib/v0.7/Serialization/src/Serialization.jl:745
deserialize at /home/shashi/code/julia7/usr/share/julia/stdlib/v0.7/Serialization/src/Serialization.jl:705
handle_deserialize at /home/shashi/code/julia7/usr/share/julia/stdlib/v0.7/Serialization/src/Serialization.jl:752
deserialize_msg at /home/shashi/code/julia7/usr/share/julia/stdlib/v0.7/Serialization/src/Serialization.jl:705
#invokelatest#1 at ./essentials.jl:691 [inlined]
invokelatest at ./essentials.jl:690 [inlined]
message_handler_loop at /home/shashi/code/julia7/usr/share/julia/stdlib/v0.7/Distributed/src/process_messages.jl:160
process_tcp_streams at /home/shashi/code/julia7/usr/share/julia/stdlib/v0.7/Distributed/src/process_messages.jl:117
#105 at ./task.jl:262
Stacktrace:
 [1] #remotecall_fetch#149 at /home/shashi/code/julia7/usr/share/julia/stdlib/v0.7/Distributed/src/remotecall.jl:379 [inlined]
 [2] remotecall_fetch(::Function, ::Distributed.Worker, ::Int64) at /home/shashi/code/julia7/usr/share/julia/stdlib/v0.7/Distributed/src/remotecall.jl:371
 [3] #remotecall_fetch#152(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Function, ::Int64, ::Int64) at /home/shashi/code/julia7/usr/share/julia/stdlib/v0.7/Distributed/src/remotecall.jl:392
 [4] remotecall_fetch at /home/shashi/code/julia7/usr/share/julia/stdlib/v0.7/Distributed/src/remotecall.jl:392 [inlined]
 [5] foo() at /home/shashi/.julia/dev/ParReviseExample/src/ParReviseExample.jl:7
 [6] top-level scope at none:0
```

This of course assumes you load Revise from your startup script.

@JeffBezanson you might find this interesting.

**Comment by shashi** [comment]
Just realized, so here the anonymous function is defined on the master process (process 1), serialized and sent to the workers. When `Revise` reloads the module, it re-writes the function on master, but couldn't do it on the workers?

**Comment by timholy** [comment]
I'm not sure it's easy to ensure the same anonymous function ends up having the same `gensymmed` name on all worker processes. Jeff would know more. Meanwhile in #157 I at least documented the obvious workaround.

**Comment by timholy** [comment]
An issue that doesn't require an explicit anonymous function (one that arises from macros):
https://stackoverflow.com/questions/57559113/problem-with-package-revise-and-distributed-loop

**Comment by timholy** [comment]
Fix requires Julia 1.3 or higher
